modyn/config/schema/pipeline/trigger/drift.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import annotations
+
+from typing import Any, ForwardRef, Literal, Optional
+
+from modyn.config.schema.base_model import ModynBaseModel
+from pydantic import Field
+
+__TriggerConfig = ForwardRef("TriggerConfig", is_class=True)
+
+
+class DataDriftTriggerConfig(ModynBaseModel):
+    id: Literal["DataDriftTrigger"] = Field("DataDriftTrigger")
+
+    detection_interval: Optional[__TriggerConfig] = Field(  # type: ignore[valid-type]
+        None, description="The Trigger policy to determine the interval at which drift detection is performed."
+    )  # currently not used
```

Comment:
I don't get this yet. How do you want to pass a trigger_config to this (and why)? can i pass a drift trigger to a drift trigger? could you elaborate a bit here?

Reply:
I noticed that drift the trigger only supports to be evaluated after every n-th sample. However, there's no real reason to not use a time schedule for assessing the drift trigger. Consequently, I wanted to support specifying a TimeTrigger or DataAmountTrigger. However, there's no real reason not to support other trigger types to as they follow the same interface. In practise you would want to use a cheaper policy though to determine the evaluation of the more expensive ones.